pkg/operator/periodicbackupcontroller/periodicbackupcontroller.go

```diff
@@ -318,3 +334,114 @@ func newCronJob() (*batchv1.CronJob, error) {
+					Labels: map[string]string{
+						backupDSLabelKey: backupDSLabelValue,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Tolerations: []corev1.Toleration{
+						{
+							Key:    masterNodeSelector,
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+					},
+					NodeSelector: map[string]string{masterNodeSelector: ""},
+					Volumes: []corev1.Volume{
+						{Name: etcdDataDirVolName, VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: etcdDataDirVolPath,
+								Type: ptr.To(corev1.HostPathUnset)}}},
+
+						{Name: etcdConfigDirVolName, VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: etcdConfigDirVolPath}}},
+
+						{Name: etcdAutoBackupDirVolName, VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: etcdAutoBackupDirVolPath}}},
+
+						{Name: etcdCertDirVolName, VolumeSource: corev1.VolumeSource{
+							HostPath: &corev1.HostPathVolumeSource{
+								Path: "/etc/kubernetes/static-pod-resources/etcd-certs"}}},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:    etcdBackupServerContainerName,
+							Image:   os.Getenv("OPERATOR_IMAGE"),
+							Command: []string{"cluster-etcd-operator", "backup-server"},
+							Env: []corev1.EnvVar{
+								{
+									Name: nodeNameEnvVar,
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: nodeNameFieldPath,
+										},
+									},
+								},
+							},
+							Args: constructBackupServerArgs(cr, strings.Join(endpoints, ",")),
+							VolumeMounts: []corev1.VolumeMount{
+								{Name: etcdDataDirVolName, MountPath: etcdDataDirVolPath},
+								{Name: etcdConfigDirVolName, MountPath: etcdConfigDirVolPath},
+								{Name: etcdAutoBackupDirVolName, MountPath: etcdAutoBackupDirVolPath},
+								{Name: etcdCertDirVolName, MountPath: etcdCertDirVolPath},
+							},
+							TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+							ImagePullPolicy:          corev1.PullIfNotPresent,
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: ptr.To(true),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return &ds
+}
+
+func getEtcdEndpoints(ctx context.Context, client kubernetes.Interface) ([]string, error) {
+	etcdEndPointsCM, err := client.CoreV1().ConfigMaps(operatorclient.TargetNamespace).Get(ctx, etcdEndpointConfigMapName, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var endpoints []string
+	for _, v := range etcdEndPointsCM.Data {
+		ep := v + etcdClientPort
+		endpoints = append(endpoints, ep)
+	}
+
+	slices.Sort(endpoints)
+	return endpoints, nil
+}
+
+func constructBackupServerArgs(cr backupv1alpha1.Backup, endpoints string) []string {
+	backupConfig := backuphelpers.NewDisabledBackupConfig()
+	backupConfig.SetBackupSpec(&cr.Spec.EtcdBackupSpec)
+	argList := backupConfig.ArgList()
```

Comment:
are you going to cleanup all related stuff in the targetConfigController in a later PR?

Reply:
yes exactly, so @hasbro17 requested to update the enhancement  and present both approaches with Pros&Cons

Then we get Consensus over which one to use 

Once the DS being approved, which I also vote for btw :D, I will clean up all the changes related to the StaticPod approach 

wdyt ?